src/main/java/book/store/onlinebookstore/controller/AuthenticationController.java

```diff
@@ -0,0 +1,44 @@
+package book.store.onlinebookstore.controller;
+
+import book.store.onlinebookstore.dto.user.UserLoginRequestDto;
+import book.store.onlinebookstore.dto.user.UserLoginResponseDto;
+import book.store.onlinebookstore.dto.user.UserRegistrationRequestDto;
+import book.store.onlinebookstore.dto.user.UserRegistrationResponseDto;
+import book.store.onlinebookstore.exception.UserRegistrationException;
+import book.store.onlinebookstore.security.AuthenticationService;
+import book.store.onlinebookstore.service.UserService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@Validated

```

Comment:
"you may also validate such things as @PathVariable, @RequestParam etc.

it will look like: @PathVariable @Positive Long id

however in order to do so you need to annotate your controller with @validated annotation

this is just FYI :)"

Said Nazar